src/Filter/UrlSectionFilter.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+namespace Phergie\Irc\Plugin\React\Url\Filter;
+
+use Phergie\Irc\Event\EventInterface;
+use Phergie\Irc\Plugin\React\EventFilter\FilterInterface;
+
+class UrlSectionFilter implements FilterInterface
+{
+    /**
+     * @var string
+     */
+    protected $value;
+
+    /**
+     * @var string
+     */
+    protected $section;
+
+    /**
+     *
+     * @param string $section
+     * @param string $value
+     */
+    public function __construct($section, $value)
+    {
+        $this->section = $section;
+        $this->value = $value;
+    }
+
+    /**
+     * @param EventInterface $event
+     * @return bool|null
+     */
+    public function filter(EventInterface $event)
+    {
+        if (!($event instanceof UrlEvent)) {
+            return null;
+        }
+
+        $section = $event->getUrlSection($this->section);
+        if ($section === false) {
+            return null;
```

Comment:
Yes if the part of the URL is missing it doesn't fall within scope of the filter. How ever in some cases it does make sense to return false if it is missing so I've added a strict bool in 2aed4c3
.